sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Utils.scala

```diff
@@ -57,4 +60,21 @@ private[sql] object DataSourceV2Utils extends Logging {
       case _ => Map.empty
     }
   }
+
+  def loadTableWithUserSpecifiedSchema(
+      provider: TableProvider,
+      schema: StructType,
+      options: CaseInsensitiveStringMap): Table = {
+    provider match {
+      case s: SupportsSpecifiedSchemaPartitioning =>
+        // TODO: `DataFrameReader`/`DataStreamReader` should have an API to set user-specified
+        //       partitioning.
```

Comment:
This doesn't make sense to me. Partitioning should be passed here. That's what this method is for.

Reply:
But there is no API to specify the partitioning here, neither `DataFrameReader` or `DataStreamReader`.

Reply:
If we are fixing the sources later, maybe we can remove this as well and fix it in the follow-up when the partition inference method is done.